subtests/docker_cli/events/events.py

```diff
@@ -98,42 +55,88 @@ def is_dupe_event(needle, haystack):
     return False
 
 
+def parse_event_docker_110(line):
+    """
+    Try to parse input as a docker 1.10 event
+    """
+    # eg <timestamp> container start <sha> (details)
+    event_re = re.compile(r'^(?P<timestamp>{timestamp})'
+                          r'\s+(?P<object>\w+)'
+                          r'\s+(?P<operation>{operation})'
+                          r'\s+(?P<identifier>{cid}|{fqin})'
+                          r'\s+\((?P<rest>.*)\)'.format(**regexes))
+    mobj = event_re.match(line)
+    if mobj is None:
+        return None
+
+    # Matched! Extract the positional fields, then try looking for source img
+    details = {
+        'datetime':   DockerTime(mobj.group('timestamp')),
+        'identifier': mobj.group('identifier'),
+        'object':     mobj.group('object'),
+        'operation':  mobj.group('operation'),
+        'source':     None,
+    }
+    # TODO: (maybe): split out components of the parenthesized list.
+    # If so, keep in mind that you can't just split on commas (because
+    # of "Red Hat, Inc.") and that the fields are output in unpredictable
+    # order: even two consecutive event lines will have different ordering.
+    source_re = re.compile(r'(^|\s)image=(?P<image>\S+)(,|$)')
+    mobj2 = source_re.search(mobj.group('rest'))
+    if mobj2 is not None:
+        details['source'] = mobj2.group('image')
+    return details
+
+
+def parse_event_docker_109(line):
+    """
+    Try to parse input as a docker < 1.10 event
+    """
+    # eg <timestamp> <sha> (from <source>) start
+    event_re = re.compile(r'^(?P<timestamp>{timestamp})'
+                          r'\s+(?P<identifier>{cid}|{fqin}):'
+                          r'(\s+\(from (?P<source>{source})\))?'
+                          r'\s+(?P<operation>{operation})'.format(**regexes))
+    mobj = event_re.match(line)
+    if mobj is not None:
+        return {
+            'datetime':   DockerTime(mobj.group('timestamp')),
+            'identifier': mobj.group('identifier'),
+            'source':     mobj.group('source'),
+            'operation':  mobj.group('operation'),
+        }
+    return None
+
+
 def parse_event(line):
     """
-    Return tuple(CID/FQIN, {DETAILS}) from parsing line
+    Return {DETAILS} from parsing line
 
     :param line: String-like containing a single event line
-    :returns: (CID/FQIN, {DETAILS}) from parsing line or None if unparseable
+    :returns: {DETAILS} from parsing line or None if unparseable
     """
```

Comment:
Those curly braces won't mess with ``str.format()`` will they?  I suppose probably not.